medical_turn_management/models/medical_turn_specialty.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2020 Creu Blanca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+from dateutil import tz
+from pytz import utc
+from datetime import time, datetime, timedelta
+
+
+class MedicalTurnSpecialty(models.Model):
+    _name = 'medical.turn.specialty'
+    _description = 'Turn Specialty'  # TODO
+
+    name = fields.Char(required=True)
+    active = fields.Boolean(
+        default=True,
+    )
+    rule_ids = fields.One2many(
+        'medical.turn.specialty.rule',
+        inverse_name='turn_specialty_id',
+    )
+
+    def _execute_rules(self, start_date, end_date):
+        results = self.env['medical.turn']
+        rules = self.mapped('rule_ids')
+        for day in range(0, (end_date-start_date).days):
+            date = start_date + timedelta(days=day)
+            for rule in rules:
+                if date.weekday() != int(rule.dayofweek):
```

Comment:
Can be done with a lambda function

Reply:
Faster this, way, If we use lambda, we will do two fors. Now it's just one.